grpc/server/src/test/java/io/helidon/grpc/server/GrpcServiceTest.java

```diff
@@ -511,9 +512,22 @@ public void update(ServiceDescriptor.Rules rules) {
      */
     private static class SynchronizedObserver<T>
             extends TestStreamObserver<T> {
+
+        private final Semaphore nextAccess = new Semaphore(1, true);
+
         @Override
-        public synchronized void onNext(T t) {
-            super.onNext(t);
+        public void onNext(T t) {
+            accessNext(() -> super.onNext(t));
+        }
+
+        private void accessNext(Runnable operation) {
+            try {
+                nextAccess.acquire();
+                operation.run();
+                nextAccess.release();
```

Comment:
This release will not happen if `operation.run()` fails.

Reply:
Fixed. Using lock instead.